### PR TITLE
Possible fix for "known" offers getting reported as unknown and some beautifyings

### PR DESCRIPTION
**Pull request by kinsi55**

### CODE REVIEW

**Comment by kinsi55** [comment]
Seeing as he is currently "modernizing" the lib overall w/ ES6 and more i thought its stuff that fits anways. Also i talked w/ him earlier today about me implementing the core of this PR. I doubt he will dislike the other changes.

**Comment by ylms** [comment]
@kinsi55 can you link me your Steam acc I want to talk with you if possible.

**Comment by kinsi55** [comment]
@ylms about what exactly?

**Comment by ylms** [comment]
@kinsi55 steam bot / programming

File: lib/classes/TradeOffer.js [hunk 1]
```diff
@@ -477,11 +474,13 @@ TradeOffer.prototype.send = function(callback) {
 	}, (err, response, body) => {
```
Comment (DoctorMcKay) [inline lib/classes/TradeOffer.js]: Why not just put `_pendingOfferSendResponses--` right here, at the top of the callback?

File: lib/index.js [hunk 2]
```diff
@@ -64,7 +65,7 @@ function TradeOfferManager(options) {
 	}
 }
 
-TradeOfferManager.prototype.setCookies = function(cookies, callback) {
+TradeOfferManager.prototype.setCookies = (cookies, callback) => {
```
Comment (DoctorMcKay) [inline lib/index.js]: I prefer to not do arrow functions for member functions, as `this` and `arguments` can be needed. In fact, what *is* the value of `this` when you do this?

File: lib/classes/TradeOffer.js (hunk 1 above)
Comment (kinsi55) [inline lib/classes/TradeOffer.js]: To possibly work around the race condition where its set to 0 while the poll is going on, but the poll reaches the unknown-check before the polldata is further updated.

I have no idea how likely this is to happen, or if it even can happen, but that was my thoughts.

Comment (DoctorMcKay) [inline lib/classes/TradeOffer.js]: Node is single-threaded, so that isn't possible.

File: lib/assets.js [hunk 3]
```diff
@@ -22,7 +22,7 @@ TradeOfferManager.prototype._digestDescriptions = function(descriptions) {
 			return;
 		}
 
-		cache[item.appid + '_' + item.classid + '_' + (item.instanceid || '0')] = item;
+		cache[`{item.appid}_{item.classid}_{item.instanceid || '0'}`] = item;
```
Comment (DoctorMcKay) [inline lib/assets.js]: Don't you need dollar signs here?

File: lib/assets.js [hunk 4]
```diff
@@ -37,7 +37,7 @@ TradeOfferManager.prototype._mapItemsToDescriptions = function(appid, contextid,
 		item.appid = appid || item.appid;
 		item.contextid = contextid || item.contextid;
 
-		var key = item.appid + '_' + item.classid + '_' + (item.instanceid || '0');
+		var key = `${item.appid}_{item.classid}_${item.instanceid || '0'}`;
```
Comment (DoctorMcKay) [inline lib/assets.js]: And here, in the second item.

File: lib/assets.js (hunk 3 above)
Comment (kinsi55) [inline lib/assets.js]: Good catch, actually you do.

**Comment by ylms** [comment]
Would it fix missing trade offer issue?

File: lib/assets.js (hunk 4 above)
Comment (DoctorMcKay) [inline lib/assets.js]: Still needs a dollar sign here.

**Comment by DoctorMcKay** [comment]
:+1: Thanks

**Pull request merged**